ui/raidboss/data/06-ew/trial/zodiark-ex.ts

```diff
@@ -19,6 +192,401 @@ const triggerSet: TriggerSet<Data> = {
+      },
+    },
+    {
+      id: 'ZodiarkEx Arcane Sigil Start Cleanup',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67E4', '67E5', '67E6'], source: 'Arcane Sigil', capture: false }),
+      delaySeconds: 0.2,
+      suppressSeconds: 0.2,
+      alertText: (data, _matches, output) => {
+        const activeFrontSigils = data.activeFrontSigils;
+        data.activeFrontSigils = [];
+        if (activeFrontSigils.length === 1 && activeFrontSigils[0]?.typeId === '67E4')
+          return output.sides!();
+        if (activeFrontSigils.length === 2 && activeFrontSigils[0]?.typeId === '67E4' && activeFrontSigils[1]?.typeId === '67E4')
+          return output.middle!();
+        if (activeFrontSigils.length === 3) {
+          for (const sig of activeFrontSigils) {
+            // Find the middle sigil
+            if (sig.x > 90 && sig.x < 110) {
+              if (sig.typeId === '67E4')
+                return output.frontsides!();
+              if (sig.typeId === '67E5')
+                return output.backmiddle!();
+              if (sig.typeId === '67E6')
+                return output.frontmiddle!();
+            }
+          }
+        }
+      },
+      outputStrings: {
+        frontsides: {
+          en: 'front sides',
+        },
+        backmiddle: {
+          en: 'back middle',
+        },
+        frontmiddle: {
+          en: 'front middle',
+        },
+        sides: Outputs.sides,
+        middle: Outputs.middle,
+      },
+    },
+    {
+      id: 'ZodiarkEx Explosion',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67E7', source: 'Zodiark' }),
+      alertText: (data, matches, output) => {
+        if (data.activeExplosions.length < 7)
+          data.activeExplosions.push({ x: parseFloat(matches.x), y: parseFloat(matches.y) });
+        if (data.activeExplosions.length === 7) {
+          const activeExplosions = data.activeExplosions;
+          data.activeExplosions = [];
+          // Explosions fall in a 9x9 grid
+          // Find out which spots are safe
+          const grid: boolean[] = [true, true, true, true, true, true, true, true, true];
+          for (const ex of activeExplosions) {
+            // Which y row is this explosion in?
+            let offset = -1;
+            if (ex.y < 90)
+              offset = 0;
+            if (ex.y > 90 && ex.y < 110)
+              offset = 3;
+            if (ex.y > 110)
+              offset = 6;
+            // Which x column is this explosion in?
+            if (ex.x < 90)
+              grid[offset + 0] = false;
+            if (ex.x > 90 && ex.x < 110)
+              grid[offset + 1] = false;
+            if (ex.x > 110)
+              grid[offset + 2] = false;
+          }
+          // FIXME: Commented parts prefer straight movement, loop prefers closest to northwest
```

Comment:
I noticed that, the code that prefers straight movement is commented for now, if I'll keep working on this depends on if and how I can make the star callouts early enough to matter. (Also the move straight thingie doesn't consider melee range but prefers straight at all costs)

I hope this loop will always yield a working answer at least I didn't really check that because I noticed that all of this is too late anyways.